spec/waiter_spec.rb

```diff
@@ -29,4 +29,42 @@
       expect(duration).to be_within(0.1).of(timeout)
     end
   end
+
+  describe '.wait_with_capybara (or_raise)' do
+    it 'raises the given error if the block is not truthy' do
+      expect do
+        described_class.wait_with_capybara(or_raise: 'BadError') { nil }
+      end.to raise_error('BadError')
+    end
+
+    it 'does not raise on failure if the "or_raise" argument is not passed' do
+      expect do
+        described_class.wait_with_capybara(timeout: 0.1) { false }
+      end.to_not raise_error
+    end
+
+    it 'returns the results from Capybara' do
+      expect(described_class.wait_with_capybara(or_raise: 'BadError') { 'result' }).to eq('result')
```

Comment:
Can we pop this on multi-line so we're nearing line-compliance for rubocop (I know it's set higher atm)